src/weights.jl

```diff
@@ -9,20 +10,21 @@ and stores the `values` (`V<:RealVector`) and `sum` (`S<:Real`).
 """
 macro weights(name)
     return quote
-        mutable struct $name{S<:Real, T<:Real, V<:AbstractVector{T}} <: AbstractWeights{S, T, V}
-            values::V
+        mutable struct $name{S<:Real, T<:Real} <: AbstractWeights{S, T}
+            values::AbstractVector{T}
```

Comment:
This introduces a type instability: the parameter needs to be kept.

Reply:
Why so? I never understood it.

Reply:
`AbstractVector` isn't a concrete type, so...